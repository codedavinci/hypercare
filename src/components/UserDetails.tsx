import { User } from "../types";
import { Avatar } from "./common";

type UserDetailsProps = {
  user: User;
};

const UserDetails = ({ user }: UserDetailsProps) => {
  const {
    avatar,
    description,
    email,
    firstname,
    join_date,
    role,
    lastname,
    username,
  } = user;

  return (
    <div className="bg-base-100">
      <figure className="flex flex-col justify-center ml-6 p-2">
        <Avatar
          src={avatar}
          altText={`${firstname} ${lastname}' profile photo.`}
        />
        <b className="mt-4 badge badge-primary bold">@{username}</b>
      </figure>
      <div className="card-body">
        <div className="flex items-center">
          <h2 className="card-title text-2xl font-bold">
            {firstname} {lastname}
          </h2>
          <div className="ml-3 badge badge-secondary">{role}</div>
        </div>

        <p className="text-gray-700 mb-4">{description}</p>
        <div className="card-actions justify-end">
          <p className="text-gray-500">
            Starting Date: {new Date(join_date).toLocaleDateString()}
          </p>
          <p className="text-gray-500">Email: {email}</p>
        </div>
      </div>
    </div>
  );
};

UserDetails.displayName = "UserDetails";

export default UserDetails;
