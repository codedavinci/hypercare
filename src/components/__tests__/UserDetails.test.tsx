import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import UserDetails from "../UserDetails";

import RAW_USERS from "../../api/__tests__/usersMock.json";

const SAMPLE_USER = RAW_USERS[0];

describe("UserDetails Component", () => {
  it("should render correctly and match snapshot", () => {
    const { asFragment } = render(<UserDetails user={SAMPLE_USER} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render user details correctly", () => {
    const { getByText } = render(<UserDetails user={SAMPLE_USER} />);
    expect(getByText("@johndoe")).toBeInTheDocument();
    expect(getByText("user")).toBeInTheDocument(); // Assuming the role is displayed in lowercase
    expect(getByText("A sample user")).toBeInTheDocument();
    expect(getByText("Email: johndoe@example.com")).toBeInTheDocument();
    expect(
      getByText(
        `Starting Date: ${new Date(SAMPLE_USER.join_date).toLocaleDateString()}`
      )
    ).toBeInTheDocument();
  });
});
