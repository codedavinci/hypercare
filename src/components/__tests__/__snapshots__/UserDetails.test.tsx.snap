// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserDetails Component should render correctly and match snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-base-100"
  >
    <figure
      class="flex flex-col justify-center ml-6 p-2"
    >
      <div
        class="avatar"
      >
        <div
          class="w-24 rounded-full ring ring-primary ring-offset-base-100 ring-offset-2"
        >
          <img
            alt="John Doe' profile photo."
            src="avatar1.png"
          />
        </div>
      </div>
      <b
        class="mt-4 badge badge-primary bold"
      >
        @johndoe
      </b>
    </figure>
    <div
      class="card-body"
    >
      <div
        class="flex items-center"
      >
        <h2
          class="card-title text-2xl font-bold"
        >
          John Doe
        </h2>
        <div
          class="ml-3 badge badge-secondary"
        >
          user
        </div>
      </div>
      <p
        class="text-gray-700 mb-4"
      >
        A sample user
      </p>
      <div
        class="card-actions justify-end"
      >
        <p
          class="text-gray-500"
        >
          Starting Date: 12/31/2022
        </p>
        <p
          class="text-gray-500"
        >
          Email: johndoe@example.com
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
